### DISCUSSION TITLE
From <x-layout.index> to <x-layout>

### DISCUSSION THREAD

**Discussion by marknuyens** [laravel/framework#37221] [Ideas] [answered]
At the time of this writing, it seems Blade does not support automatic name resolving when it comes to having one single index file in a directory.

To illustrate, please see the example below:

File path: `resources/views/components/layout/index.blade.php`
Blade syntax: `<x-layout.index />`

Wouldn't it be nice to have support for this following instead:

Blade syntax: `<x-layout />`

Let me know your thoughts!

Thanks.

**Comment by aguingand** [comment]
What about if both `resources/views/components/layout/index.php` and `resources/views/components/layout.php` exists ?

**Reply by dennisprudlo** [reply, marked as answer]
This is already in the framework: https://github.com/laravel/framework/pull/38847

**Reply by marknuyens** [reply]
Oke, great 🤩
